File: Weaver/app/schemas/dataset.py
from pydantic import BaseModel


class ColumnSchema(BaseModel):
    name: str
    inferred_type: str
    missing_count: int
    unique_count: int
    sample_values: list[str | int | float]


class DatasetUploadResponse(BaseModel):
    dataset_id: str
    filename: str
    row_count: int
    column_count: int
    columns: list[ColumnSchema]
    preview_rows: list[dict[str, str | int | float | None]]


class DatasetDetails(BaseModel):
    dataset_id: str
    filename: str
    row_count: int
    column_count: int
    columns: list[ColumnSchema]
    preview_rows: list[dict[str, str | int | float | None]]

File: Kolam/src/types/dataset.ts
export interface ColumnSchema {
  name: string;
  inferred_type: string;
  missing_count: number;
  unique_count: number;
  sample_values: (string | number)[];
}

export interface DatasetUploadResponse {
  id: string;
  filename: string;
  row_count: number;
  column_count: number;
  columns: ColumnSchema[];
  preview_rows: Record<string, string | number | null>[];
}

export interface DatasetDetails {
  id: string;
  filename: string;
  row_count: number;
  column_count: number;
  columns: ColumnSchema[];
  preview_rows: Record<string, string | number | null>[];
}
